test/neural_de.jl

```diff
@@ -154,6 +189,33 @@ grads = Zygote.gradient(()->sum(node(x)),Flux.params(x,node))
     @test_broken grads = Zygote.gradient(()->sum(node(xs)),Flux.params(xs,node)) isa Tuple
     @test_broken ! iszero(grads[xs])
     @test_broken ! iszero(grads[node.p])
+
+    node = NeuralODE(fastcdudt,tspan,Tsit5(),save_everystep=false,save_start=false)
+    grads = Zygote.gradient(()->sum(node(x)),Flux.params(x,node))
+    @test ! iszero(grads[x])
+    @test ! iszero(grads[node.p])
+
+    @test_broken grads = Zygote.gradient(()->sum(node(xs)),Flux.params(xs,node)) isa Tuple
+    @test_broken ! iszero(grads[xs])
+    @test_broken ! iszero(grads[node.p])
+
+    node = NeuralODE(fastcdudt,tspan,Tsit5(),saveat=0.0:0.1:1.0)
+    grads = Zygote.gradient(()->sum(node(x)),Flux.params(x,node))
+    @test ! iszero(grads[x])
+    @test ! iszero(grads[node.p])
+
+    @test_broken grads = Zygote.gradient(()->sum(node(xs)),Flux.params(xs,node)) isa Tuple
+    @test_broken ! iszero(grads[xs])
+    @test_broken ! iszero(grads[node.p])
+
+    node = NeuralODE(fastcdudt,tspan,Tsit5(),saveat=0.1)
+    grads = Zygote.gradient(()->sum(node(x)),Flux.params(x,node))
+    @test ! iszero(grads[x])
+    @test ! iszero(grads[node.p])
+
+    @test_broken grads = Zygote.gradient(()->sum(node(xs)),Flux.params(xs,node)) isa Tuple
+    @test_broken ! iszero(grads[xs])
+    @test_broken ! iszero(grads[node.p])

```

Comment:
It would be easier to read if the cached test is paired right next to the uncached test